cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/system/CDAPLogAppender.java

```diff
@@ -66,7 +66,15 @@ public class CDAPLogAppender extends AppenderBase<ILoggingEvent> implements Flus
   private int fileRetentionDurationDays;
   private int fileCleanupBatchSize;
 
+  private final boolean disableLogCleanerForTest;
+
+  @VisibleForTesting
```

Comment:
nit: add javadoc here explaining why this is needed

Reply:
Done